### DISCUSSION TITLE
touchscreen has "limited speed"?

### DISCUSSION THREAD

**Discussion by soliwoli** [ppy/osu#22763] [Q&A] [answered]
when im directly on the hit circle its easy to hit but when the cursor far away from the circle when i hit the input is delayed?
using linux btw

**Comment by Bruno5430** [comment, marked as answer]
It depends on your distro and your configuration mostly
Also you should provide a video showing exactly what's going on (touchscreen has "limited speed" is not very informative)

Generally some factors that can increase latency in linux:
- If using wayland, lazer will default to use the x11 backend through xwayland, increasing latency
Lazer can use a wayland backend if you set `SDL_VIDEODRIVER=wayland` as a launch argument
- Sound latency can be very high on most out-of-the-box distros, so, unless you manually configured it, it will be a problem even if you set the offset in lazer, as hitsounds will no longer match with the song and the rythm
- If you didn't change the framerate limiter of the game, it will force the system to use vsync and will increase latency a lot, even if lazer manages input without relying on framerate

**Reply by soliwoli** [reply]
how o set a wayland backend?

**Reply by Bruno5430** [reply]
If you are running lazer directly from the appimage you could [register the appimage as a desktop app](https://askubuntu.com/questions/902672/registering-appimage-files-as-a-desktop-app), then edit the desktop file to [set the environment variable directly from the desktop file](https://askubuntu.com/questions/144968/set-variable-in-desktop-file), and then [refresh the desktop entries](https://wiki.archlinux.org/title/Desktop_entries#Update_database_of_desktop_entries) so the entry will appear without restarting the whole machine

this would be an example for a `.desktop` file for lazer
```

[Desktop Entry]
Type=Application
Name=osu!lazer
Comment=Rhythm is just a *click* away!
Icon=/path/to/icon.png
Exec=env SDL_VIDEODRIVER=wayland /path/to/osu.AppImage
Terminal=false
Categories=Game
```

**Comment by soliwoli** [comment]
https://user-images.githubusercontent.com/122650721/222582621-3181e7fd-8ca8-4172-a11b-56a1104db4d0.mp4

**Reply by soliwoli** [reply]
see how its delayed when your far from the block but when your near it its fine

**Reply by soliwoli** [reply]
im on unlimited framerate btw

**Reply by Bruno5430** [reply]
are you using x11 or wayland?
which de/wm are you using?

**Reply by Bruno5430** [reply]
did you set any environment variable for lazer?

**Reply by soliwoli** [reply]
wayland
i dont know what you mean by de/wm
i did not set a env variable

**Reply by Bruno5430** [reply]
de/wm means desktop enviroment or window manager

**Comment by Bruno5430** [comment]
You are running lazer through xwayland, which increases latency so much it becomes ridiculous (as you can see in your gameplay)
Try what was proposed before
> - If using wayland, lazer will default to use the x11 backend through xwayland, increasing latency
Lazer can use a wayland backend if you set `SDL_VIDEODRIVER=wayland` as a launch argument

This was already discussed in #22651
